okhttp/src/main/java/okhttp3/internal/ws/WebSocketReader.kt

```diff
@@ -69,6 +70,9 @@ class WebSocketReader(
   private val controlFrameBuffer = Buffer()
   private val messageFrameBuffer = Buffer()
 
+  /** Lazily initialized on first use. */
+  private var messageInflater: MessageInflater? = null
```

Comment:
FWIW I still prefer

```
    private val messageInflater = lazy(mode = LazyThreadSafetyMode.NONE) { MessageInflater(noContextTakeover) }
```

Reply:
That allocates and causes Kotlin to generate a `KProperty[]` and eagerly initialize all of the `KProperty` instances on classload. Very bad.

Reply:
OK, so not the type of delegate, but the use of a delegate full stop.  Thanks

https://kotlinlang.org/docs/reference/delegated-properties.html#translation-rules

Reply:
For now. They're working on eliminating the `KProperty[]` and `KProperty`, but you'll still have to allocate the delegate instance, though that's not too bad.